Source/WebKit/Platform/cocoa/ExtensionCapability.h

```diff
@@ -28,19 +28,24 @@
 #if ENABLE(EXTENSION_CAPABILITIES)
 
 #include <wtf/Forward.h>
+#include <wtf/RetainPtr.h>
 
-OBJC_CLASS _SECapability;
+OBJC_CLASS BEProcessCapability;
 
 namespace WebKit {
 
 class ExtensionCapability {
 public:
     virtual ~ExtensionCapability() = default;
     virtual String environmentIdentifier() const = 0;
-    virtual RetainPtr<_SECapability> platformCapability() const = 0;
+    BEProcessCapability *platformCapability() const { return m_platformCapability.get(); }
+    void setPlatformCapability(BEProcessCapability *capability) { m_platformCapability = capability; }
```

Comment:
Why not pass it to the constructor instead?

Reply:
A setter only makes sense if you need to update the value later on, after construction. However, it doesn't seem you do so passing it to the constructor would make more sense.

Reply:
That is a good point. I tried doing this, and in the case when we're initializing MediaCapability, we need to initialize m_mediaEnvironment first, which may require us to have a setter. Given this, would you be ok with keeping the setter?

Reply:
Ok but please mark it as `protected` instead of `public` at least.

Reply:
Ah, good call, thank you!